dependency-management/dependency-management-impl/src/main/java/io/cloudslang/dependency/impl/services/DependencyServiceImpl.java

```diff
@@ -200,6 +233,37 @@ private void buildDependencyFile(String[] gav) {
             throw new IllegalStateException(fileToReturn.getPath() + " not found");
         }
         appendSelfToPathFile(gav, fileToReturn);
+        sendMavenDependencyBuildFinishedEvent(gav);
+    }
+
+    private void sendMavenDependencyBuildFinishedEvent(String[] gav) {
+        String message = "Artifact download complete";
+        if (gav.length >= 3) {
+            message = String.format("Downloading complete for artifact with gav: %s:%s:%s ", gav[0], gav[1], gav[2]);
+        }
+        Map<String, Serializable> data = new HashMap<>();
+        data.put(EventConstants.MAVEN_DEPENDENCY_BUILD_FINISHED, message);
+        ScoreEvent eventWrapper = new ScoreEvent(EventConstants.MAVEN_DEPENDENCY_BUILD_FINISHED, (Serializable) data);
+        dispatchEvent(eventWrapper);
+    }
+
+    private void sendMavenDependencyBuildEvent(String[] gav) {
+        String message = "Downloading artifact..";

```

Comment:
maybe you can simplify the logic by always printing everything from the `gav `array joined by : character so that the message is not just Download artifact when the length is less than 3